sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdInfoLiveData.java

```diff
@@ -0,0 +1,477 @@
+         *               TransferInfo LiveData source to know
+         */
+        private Result(@NonNull MutableLiveData<TransferIdOrError> transferIdOrErrorLiveData,
+                       @NonNull LiveData<TransferInfo> transferInfoLiveData,
+                       @NonNull TransferFlags transferFlags) {
+            this.liveDataPair = new LiveDataPair(transferIdOrErrorLiveData, transferInfoLiveData);
+            this.transferFlags = transferFlags;
+        }
+
+        /**
+         * Get the pair of holding transferIdOrError and TransferInfo LiveData.
+         *
+         * @return the LiveData pair
+         */
+        LiveDataPair getLiveDataPair() {
+            return this.liveDataPair;
+        }
+
+        /**
+         * Get the shared object that the TransferInfo LiveData source check for any flag
+         * set by TransferClient methods on this transfer.
+         *
+         * @return the object that TransferClient methods update any flag that it want
+         * TransferInfo LiveData source to know
+         */
+        TransferFlags getTransferFlags() {
+            return this.transferFlags;
+        }
+    }
+
+    /**
+     * Instance of this type is used by TransferClient methods to set any flag that it want
+     * TransferInfo LiveData source of the transfer to know.
+     *
+     * If a transfer has any 'active Observers' then that transfer's {@link TransferFlags}
+     * and LiveData pair (TransferInfo and TransferIdOrError LiveData) will be tracked
+     * in the {@link TransferIdInfoLiveDataCache} cache. By 'active Observers', we mean the
+     * Observers of TransferInfo LiveData those are not GC-ed yet hence the corresponding cache entry.
+     *
+     * The cache is shared across all {@link TransferClient} instances within a application process.
+     * This enables all Observers of a transfer [e.g. Observers of upload(tid: 1), Observers of
+     * resume(tid: 1)] to share the same source TransferInfo LiveData. Because of TransferInfo LiveData
+     * sharing, the effect of any flag set in {@link TransferFlags} by one {@link TransferClient} will be
+     * visible to Observers of same transfer in a different {@link TransferClient}.
+     */
+    final static class TransferFlags {
+        private volatile boolean userPaused;
+
+        /**
+         * Set the flag indicating that user paused a transfer by calling {@link TransferClient#pause(long)}.
+         */
+        void setUserPaused() {
+            this.userPaused = true;
+        }
+
+        /**
+         * Check whether the user paused the transfer in the current session.
+         * Note: this method reset the flag once called.
+         *
+         * @return true if user paused the transfer, false otherwise.
+         */
+        @MainThread
+        boolean isUserPaused() {
+            boolean b = this.userPaused;
+            this.userPaused = false;
```

Comment:
Why is this reset when checking its value?

Reply:
No specific reasons just to view it as one time consumable value.

Reply:
and Nothing breaks if we remove it :)

Reply:
now I remember why it is a one-time consumable value because if there is a `cancellation` after a `pause`, we don't want to treat that as a `pause`.

Reply:
Probably just add a clarifying comment. I'm sure this question will come up again in the future otherwise.